nodejs/week2/homework/meal-sharing/src/backend/api/reservations-router.js

```diff
@@ -0,0 +1,41 @@
+const express = require("express");
+const router = express.Router();
+
+const reservations = require("./../data/reservations.json");
+
+// router.get("/", async (request, response) => {
+//   try {
+//     console.log(reservations);
+//     console.log("in /api/reservations");
+//   } catch (error) {
+//     throw error;
+//   }
+// });
+
+
+// /api/reservations/{id}	Respond with the json for the reservation with 
+// the corresponding id *
+
+router.get("/:id", async (request, response) => {
+
+    var requestId =Number(request.params.id);
+  
+    const find=reservations.some(reservation=>reservation.id==requestId)
+  
+    const result=reservations.filter(reservation=>reservation.id===requestId);
```

Comment:
Same here, in this case you can also use [find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) since you're looking for one meal

Reply:
Hello , thank you for the comments and specially the  link for , formatting . I did what you said in here, but when the id does not exist, it still returns an empty array.
so I commented out `maxPriceFinder`, and use following statements ,
`if(result !==isNAN`
`if(result !== 0`
`if(result !==[]`
`if(result !=={}`
and none of them worked , each time I looked for id 9, it returned an empty array.

Reply:
What I meant here is that instead of using first .some and then filter you could do this with .find:
```
const reservation = reservations.find(reservation => reservation.id == requestId);
if (reservation){
    response.send(reservation);
}else{
    response.send(`There is no Reservation with id ${requestId}`);
}
```
So .find, as per the link, returns undefined if the id does not exist and you can check it simply by doing
 ```if (reservation)```